value/power.go

```diff
@@ -135,34 +135,36 @@ func exponential(conf *config.Config, x *big.Float) *big.Float {
 	}
 	// §4.3.1 & §4.4.2 (k ≥ 1)
 	k := int(math.Ceil(math.Sqrt(float64(conf.FloatPrec()))))
-	// added precision (§4.4)
-	prec := uint(math.Log(float64(conf.FloatPrec()))) + 1
+	// Extra precision (§4.4)
+	extra := uint(math.Log(float64(conf.FloatPrec()))) + 1
 	if -k < exp {
 		// -k <= -1 < exp
 		exp += k
 		// 0 ≤ k-1 < exp (condition needed to undo argument reduction)
 		z.SetMantExp(z, -exp)
 		// 2 bits of added precision per multiplication when undoing argument reduction.
-		prec += 2 * uint(exp)
+		extra += 2 * uint(exp)
 	}
 
 	n := new(big.Float)
-	t := newFxP(conf, prec)
-	term := newFxP(conf, prec).SetUint64(1)
-	term0 := newFxP(conf, prec)
-	sum := newFxP(conf, prec).SetUint64(1)
+	t0 := newFxP(conf, extra)
+	t1 := newFxP(conf, extra)
+	term := newFxP(conf, extra).SetUint64(1)
+	sum := newFxP(conf, extra).SetUint64(1)
 
 	// TODO: cannot use loop here since it does not handle the extended precision.
 	// term(n) = tern(n-1) × x/n is faster than term(n) = x^n / n! (saves one .Mul)
 	for i := uint64(1); ; i++ {
-		t.Quo(z, n.SetUint64(i))
-		// term.Mul(term, t) requires a temp Float for term. Manage that ourselves
-		// by using our own temp term0, then swap the pointers term and term0.
-		term0.Mul(term, t)
-		term0, term = term, term0
-		sum.Add(sum, term)
-
-		// if term < 1 ulp, we are done. This check is done after the summation since
+		t0.Quo(z, n.SetUint64(i))
+		// term.Mul(term, t) and sum.Add(sum, term) require a temp Float for the
+		// result. Manage that ourselves by using our own temps t0, t1, then swap the
+		// pointers.
+		t1.Mul(term, t0)
+		t1, term = term, t1
+		t1.Add(sum, term)
+		t1, sum = sum, t1
+
```

Comment:
That change was not requested but I missed the need for a temp Float in sum.Add(sum, ...). Also renamed all temps consistently and changed the argument reduction undo to swap pointers after .Mul for consistency.